stonesoup/models/transition/nonlinear.py

```diff
@@ -1,46 +1,215 @@
+                          x_{vel}\cos\omega dt - y_{vel}\sin\omega dt \\
+                          y+ \frac{v_{vel}}{\omega}\sin\omega dt +
+                              \frac{x_{vel}}{\omega}(1-\cos\omega dt) \\
+                          x_{vel}\sin\omega dt + y_{vel}\sin\omega dt \\
+                          \omega
+                      \end{bmatrix}
+
+        .. math::
+             Q_t & = & \begin{bmatrix}
+                          \frac{dt^4q_x^2}{4} & \frac{dt^3q_x^2}{2} & \frac{dt^4q_xq_y}{4} &
+                              \frac{dt^3q_xq_y}{2} & \frac{dt^2q_xq_\omega}{2} \\
+                          \frac{dt^3q_x^2}{2} & dt^2q_x^2 & \frac{dt^3q_xq_y}{2} & dt^2q_xq_y &
+                              dt q_x q_\omega \\
+                          \frac{dt^4q_xq_y}{4} & \frac{dt^3q_xq_y}{2} & \frac{dt^4q_y^2}{4} &
+                              \frac{dt^3q_y^2}{2} & \frac{dt^2q_y q_\omega}{2} \\
+                          \frac{dt^3q_x q_y}{2} & dt^2q_xq_y & \frac{dt^3q_y^2}{2} &
+                              dt^2q_y^2 & dt q_y q_\omega \\
+                          \frac{dt^2q_xq_\omega}{2} & dtq_xq_\omega & \frac{dt^2q_yq_\omega}{2} &
+                              dt q_y q_\omega & q_\omega^2
+                     \end{bmatrix}
     """
+    linear_noise_coeffs: np.ndarray = Property(
+        doc=r"The acceleration noise diffusion coefficients :math:`[q_x, \: q_y]^T`")
+    turn_noise_coeff: float = Property(
+        doc=r"The turn rate noise coefficient :math:`q_\omega`")
+
+    @property
+    def ndim_state(self):
+        """ndim_state getter method
+
+        Returns
+        -------
+        : :class:`int`
+            The number of combined model state dimensions.
+        """
+        return 5
 
-    def function(self, state, **kwargs) -> StateVector:
-        """Applies each transition model in :py:attr:`~model_list` in turn to the state's
-        corresponding state vector components.
-        For example, in a 3D state space, with :py:attr:`~model_list` = [modelA(ndim_state=2),
-        modelB(ndim_state=1)], this would apply modelA to the state vector's 1st and 2nd elements,
-        then modelB to the remaining 3rd element.
+    def function(self, state, noise=False, **kwargs) -> StateVector:
+        time_interval_sec = kwargs['time_interval'].total_seconds()
+        sv1 = state.state_vector
+        turn_rate = sv1[4, :]
+        # Avoid divide by zero in the function evaluation
+        turn_rate[turn_rate == 0.] = np.finfo(float).eps
+        dAngle = turn_rate * time_interval_sec
+        sv2 = StateVectors(
+            [sv1[0] + np.sin(dAngle)/turn_rate * sv1[1] - sv1[3] / turn_rate *
+             (1. - np.cos(dAngle)),
+             sv1[1] * np.cos(dAngle) - sv1[3] * np.sin(dAngle),
+             sv1[1] / turn_rate * (1. - np.cos(dAngle)) + sv1[2] + sv1[3] * np.sin(dAngle)
+             / turn_rate,
+             sv1[1] * np.sin(dAngle) + sv1[3] * np.cos(dAngle),
+             turn_rate])
```

Comment:
Should this be using `sv1[0, :]`, etc. Also, I think by pre-calculating cos/sin, you can also save some space on the lines.
```suggestion
        cos_dAngle = np.cos(dAngle)
        sin_dAngle = np.sin(dAngle)
        sv2 = StateVectors([
            sv1[0, :] + sin_dAngle/turn_rate*sv1[1, :] - sv1[3, :]/turn_rate*(1. - cos_dAngle),
            sv1[1, :]*cos_dAngle - sv1[3, :]*sin_dAngle,
            sv1[1, :]/turn_rate*(1. - cos_dAngle) + sv1[2, :] + sv1[3, :]*sin_dAngle/turn_rate,
            sv1[1, :]*sin_dAngle + sv1[3, :]*cos_dAngle,
            turn_rate
        ])
```

Reply:
@sdhiscocks In working on the ConstantTurn model I found an issue with the covariance matrix that causes a problem with the Model.pdf(). I've changed the covariance matrix used in the CT model, so that it uses the discretized form of continuous time motion - this makes it consistent with the current linear form constant turn models. This avoids the issue I mentioned.